cli/src/main/java/keywhiz/cli/commands/DeleteAction.java

```diff
@@ -102,7 +105,11 @@ public DeleteAction(DeleteActionConfig deleteActionConfig, KeywhizClient client)
               return;
             } else if (line.toUpperCase().startsWith("Y")) {
               logger.info("Deleting secret '{}'.", sanitizedSecret.name());
-              keywhizClient.deleteSecretWithId(sanitizedSecret.id());
+              if (deleteActionConfig.mode == null) {
+                keywhizClient.deleteSecretWithId(sanitizedSecret.id());
+              } else {
+                keywhizClient.deleteSecretWithId(sanitizedSecret.id(), deleteActionConfig.mode);
+              }
```

Comment:
Although we could consolidate to a single call by defaulting to soft deletion, I preserved the original invocation to maintain compatibility with existing test expectations. I'd be more than happy to do the consolidation, however.